contrib/builder/dockerfiles/generate.sh

```diff
@@ -0,0 +1,156 @@
+#!/bin/bash
+set -e
+
+# usage: ./generate.sh [versions]
+#    ie: ./generate.sh
+#        to update all Dockerfiles in this directory
+#    or: ./generate.sh debian-jessie
+#        to only update debian-jessie/Dockerfile
+#    or: ./generate.sh debian-newversion
+#        to create a new folder and a Dockerfile within it
+
+cd "$(dirname "$(readlink -f "$BASH_SOURCE")")"
+
+types=( arm gccgo ppc64le s390x )
+
+for type in "${types[@]}"; do
+	case "$type" in
+		arm)
+			from='ioft/armhf-ubuntu:14.04'
+			;;
+		gccgo)
+			from='gcc:5.2'
+			;;
+		ppc64le)
+			from='ppc64le/gcc:5.2'
+			;;
+		s390x)
+			from='s390x/gcc:5.2'
+			;;
+	esac
+
+	echo "$type -> FROM $from"
+
+	dockerfile="../../../Dockerfile.${type}"
+
+	cat > "$dockerfile" <<-EOF
+		#
+		# THIS FILE IS AUTOGENERATED; SEE "contrib/builder/dockerfiles/generate.sh"!
+		#
+		# This file describes the standard way to build Docker, using docker
+		# for ${type}.
+		#
+		# Usage:
+		#
+		# # Assemble the full dev environment. This is slow the first time.
+		# docker build -t docker -f Dockerfile.${type} .
+		#
+
+
+
+		FROM $from
+	EOF
+
+	echo >> "$dockerfile"
+
+	# this list is sorted alphabetically; please keep it that way
+	packages=(
+		apparmor
+		aufs-tools
+		btrfs-tools
+		build-essential
+		curl ca-certificates
+		git
+		iptables
+		jq
+		net-tools
+		libapparmor-dev
+		libcap-dev
+		libsqlite3-dev
+	)
+
+	echo "RUN apt-get update && apt-get install -y ${packages[*]} --no-install-recommends && rm -rf /var/lib/apt/lists/*" >> "$dockerfile"
+
+	echo >> "$dockerfile"
+
+	# install lvm2 from source
+	cat >> "$dockerfile" <<-'EOF'
+	# Get lvm2 source for compiling statically
+	RUN git clone -b v2_02_103 https://git.fedorahosted.org/git/lvm2.git /usr/local/lvm2
+	# see https://git.fedorahosted.org/cgit/lvm2.git/refs/tags for release tags
+
+	# Compile and install lvm2
+	RUN cd /usr/local/lvm2 \
+		&& ./configure --enable-static_link \
+		&& make device-mapper \
+		&& make install_device-mapper
+	# see https://git.fedorahosted.org/cgit/lvm2.git/tree/INSTALL
```

Comment:
ah yeah that would be dope :)

On Tue, Dec 8, 2015 at 2:20 PM, Tianon Gravi <notifications@github.com>
wrote:

> In contrib/builder/dockerfiles/generate.sh
> <https://github.com/docker/docker/pull/18415#discussion_r47022934>:
>
> > +	echo "RUN apt-get update && apt-get install -y ${packages[*]} --no-install-recommends && rm -rf /var/lib/apt/lists/*" >> "$dockerfile"
> > +
> > +	echo >> "$dockerfile"
> > +
> > +	# install lvm2 from source
> > +	cat >> "$dockerfile" <<-'EOF'
> > +	# Get lvm2 source for compiling statically
> > +	RUN git clone -b v2_02_103 https://git.fedorahosted.org/git/lvm2.git /usr/local/lvm2
> > +	# see https://git.fedorahosted.org/cgit/lvm2.git/refs/tags for release tags
> > +
> > +	# Compile and install lvm2
> > +	RUN cd /usr/local/lvm2 \
> > +		&& ./configure --enable-static_link \
> > +		&& make device-mapper \
> > +		&& make install_device-mapper
> > +	# see https://git.fedorahosted.org/cgit/lvm2.git/tree/INSTALL
>
> Or exclude_graphdriver_devicemapper [image: :stuck_out_tongue:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/18415/files#r47022934>.
>